### PR TITLE
refactor(server): auth

### PR DESCRIPTION
**Pull request by forehalo**

### CODE REVIEW

**Comment by forehalo** [comment]
* **#7999** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7999?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2  dependent PRs ([#8019](https://github.com/toeverything/AFFiNE/pull/8019) <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8019?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#8032](https://github.com/toeverything/AFFiNE/pull/8032) <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8032?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#7994** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7994?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @forehalo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by EYHN — APPROVED** [review]

File: packages/backend/server/src/plugins/oauth/controller.ts [hunk 1]
```diff
@@ -93,43 +104,18 @@ export class OAuthController {
...
+    const user = await this.loginFromOauth(
+      state.provider,
+      externAccount,
+      tokens
+    );
+
+    await this.auth.setCookies(req, res, user.id);
+    res.send({
+      id: user.id,
+      /* @deprecated */
```
Comment (darkskygit) [inline packages/backend/server/src/plugins/oauth/controller.ts:116]: we have change the api endpoint, is it necessary to keep the old client compatibility code?

File: packages/backend/server/src/core/auth/resolver.ts [hunk 2]
```diff
@@ -101,14 +95,6 @@ export class AuthResolver {
       throw new LinkExpired();
     }
 
-    const config = await this.config.runtime.fetchAll({
```
Comment (darkskygit) [inline packages/backend/server/src/core/auth/resolver.ts:104]: why remove password limit

File: packages/backend/server/src/core/auth/guard.ts [hunk 3]
```diff
@@ -60,70 +52,39 @@ export class AuthGuard implements CanActivate, OnModuleInit {
...
-    req: Request
-  ): Promise<{ user: CurrentUser; session: UserSession } | null> {
-    if (req.user && req.session) {
-      return {
-        user: req.user,
-        session: req.session,
-      };
+  async signIn(req: Request): Promise<Session | null> {
+    if (req.session) {
+      req.session;
```
Comment (darkskygit) [inline packages/backend/server/src/core/auth/guard.ts]: missing return

Comment (darkskygit) [reply packages/backend/server/src/core/auth/guard.ts]: prefer return with 304 code

File: packages/backend/server/src/core/auth/service.ts [hunk 4]
```diff
@@ -48,6 +25,14 @@ export function sessionUser(
   });
 }
 
+function extractTokenFromHeader(authorization: string) {
+  if (!/^Bearer\s/i.test(authorization)) {
```
Comment (Brooooooklyn) [inline packages/backend/server/src/core/auth/service.ts:29]: use String#startsWith instead

File: packages/backend/server/src/plugins/oauth/controller.ts (hunk 1 above)
Comment (forehalo) [reply packages/backend/server/src/plugins/oauth/controller.ts:116]: old clients always use `/desktop-signin` as a transfer

File: packages/backend/server/src/core/auth/guard.ts (hunk 3 above)
Comment (forehalo) [reply packages/backend/server/src/core/auth/guard.ts]: all of them are API, redirect is useless actually

File: packages/backend/server/src/core/auth/resolver.ts (hunk 2 above)
Comment (forehalo) [reply packages/backend/server/src/core/auth/resolver.ts:104]: `auth.changePassword` uses `user.updateUser` which includes all the validations

File: packages/backend/server/src/core/auth/service.ts (hunk 4 above)
Comment (forehalo) [reply packages/backend/server/src/core/auth/service.ts:29]: I don't know is there any case header passed with `Bearer xxx` and `bearer xxx` (in difference case)

Comment (darkskygit) [reply packages/backend/server/src/core/auth/service.ts:29]: toLowerCase first?

Comment (forehalo) [reply packages/backend/server/src/core/auth/service.ts:29]: that cost much more

**Review by Brooooooklyn — APPROVED** [review]

**Pull request merged**